docs/src/modules/components/ApiPage/processors/classes.ts

```diff
@@ -0,0 +1,60 @@
+import { PropsTranslations, ComponentApiContent } from '@mui-internal/api-docs-builder';
+import { Translate } from '@mui/docs/i18n';
+import kebabCase from 'lodash/kebabCase';
+import type { TableOfContentsParams } from 'docs/src/modules/components/ApiPage';
+
+export interface ClassDefinition {
+  className: string;
+  key: string;
+  hash: string;
+  description?: string;
+  isGlobal?: boolean;
+  isDeprecated?: boolean;
+  deprecationInfo?: string;
+}
+
+export type GetCssToCParams = {
+  classes: ClassDefinition[];
+  t: Translate;
+  hash?: string;
+};
+
+export const getClassesToC = ({ classes, t, hash }: GetCssToCParams): TableOfContentsParams[] =>
+  !classes || classes.length === 0
+    ? []
+    : [
+        {
+          text: t('api-docs.classes'),
+          hash: hash ?? 'classes',
+          children: [
+            ...classes.map(({ key, hash: classeHash }) => ({
+              text: key,
+              hash: classeHash,
+              children: [],
+            })),
+          ],
+        },
+      ];
+
+export interface ClassesApiProcessorParams {
+  componentClasses: ComponentApiContent['classes'];
+  classDescriptions: PropsTranslations['classDescriptions'];
+  componentName: string;
+}
+
+export function classesApiProcessor(params: ClassesApiProcessorParams): ClassDefinition[] {
```

Comment:
Effectively I was not super happy with those namings

About `mapXtoY` naming, my struggle is that the `X` is not easy to name since it's a mix of descriptions and translations. So I went with a getter naming `get[Yyy]ApiDefinitions`